doc/case-studies/timelock/token.tla

```diff
@@ -0,0 +1,43 @@
+(*
+ * SEP-41 Token interface implementation
+ * Can be used to track multiple tokens
+ *
+ * Andrey Kuprianov, 2024
+ *)
+
+---- MODULE token ----
+
+EXTENDS env
+
+VARIABLES
+    (*  
+        @type: $address -> $address -> Int;
+    *)
+    token_balances
```

Comment:
Similar to my comment in `env`, be conservative with `VARIABLES`, because they'd have to be instrumented, and I don't think this one is instrumented anywhere. Do we know how instrumentation behaves, repectively should behave, w.r.t. `soroban_sdk::contractimport!`?